### PR TITLE
Replace nonce handler feature flag with on-chain config

### PR DESCRIPTION
**Pull request by pathornteng**
Replace `IncrementNonceOnFailedTxFeature` feature flag with on-chain config so that we can toggle the new behavior of `NonceHandler`.  

go-loom: https://github.com/loomnetwork/go-loom/pull/451

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -119,7 +118,11 @@ func (n *NonceHandler) Nonce(
 		n.nonceCache = make(map[string]uint64)
 	}
 	var seq uint64
-	if state.FeatureEnabled(features.IncrementNonceOnFailedTxFeature, false) && !isCheckTx {
+
+	cfg := state.Config()
+	IncrementNonceOnFailedTx := cfg.GetNonceHandler().GetIncNonceOnFailedTx() > 0
```
Comment (enlight) [inline auth/auth.go]: `IncrementNonceOnFailedTx` -> `incrementNonceOnFailedTx`
And why do we need `cfg` if you only access it once?

File: auth/auth_test.go [hunk 2]
```diff
@@ -13,10 +13,13 @@ import (
 	loom "github.com/loomnetwork/go-loom"
 	"github.com/loomnetwork/go-loom/auth"
 	"github.com/loomnetwork/loomchain"
-	"github.com/loomnetwork/loomchain/features"
 	"github.com/loomnetwork/loomchain/store"
 )
 
+const (
+	IncrementNonceOnFailedTx = "NonceHandler.IncNonceOnFailedTx"
```
Comment (enlight) [inline auth/auth_test.go]: `IncrementNonceOnFailedTx` -> `incrementNonceOnFailedTx`
No need to export this from a test.

File: auth/auth_test.go [hunk 3]
```diff
@@ -62,7 +65,8 @@ func TestSignatureTxMiddlewareMultipleTxSameBlock(t *testing.T) {
 	ctx = context.WithValue(ctx, ContextKeyCheckTx, true)
 	kvStore := store.NewMemStore()
 	state := loomchain.NewStoreState(ctx, kvStore, abci.Header{Height: 27}, nil, nil)
-	state.SetFeature(features.IncrementNonceOnFailedTxFeature, true)
+	err = state.ChangeConfigSetting(IncrementNonceOnFailedTx, "1")
```
Comment (enlight) [inline auth/auth_test.go]: Can do `loomchain.NewStoreState(...).WithOnChainConfig(cfg)` instead, fewer hardcoded strings to deal with.

File: auth/auth_test.go [hunk 4]
```diff
@@ -153,6 +161,12 @@ func TestRevertedTxNonceMiddleware(t *testing.T) {
 	})
 	require.NoError(t, err)
 
+	nonceTxBytes3, err := proto.Marshal(&NonceTx{
```
Comment (enlight) [inline auth/auth_test.go]: Move this down, closer to where it's actually used.

File: auth/auth.go (hunk 1 above)
Comment (pathornteng) [reply auth/auth.go]: Fixed

File: auth/auth_test.go (hunk 2 above)
Comment (pathornteng) [reply auth/auth_test.go]: Fixed

File: auth/auth_test.go (hunk 3 above)
Comment (pathornteng) [reply auth/auth_test.go]: Fixed

File: auth/auth_test.go (hunk 4 above)
Comment (pathornteng) [reply auth/auth_test.go]: Fixed

**Pull request merged**